native: `init_tcp_socket()` maybe returns uninitialized variable `s`

```
make[2]: Entering directory `/home/kijewski/RIOT/RIOT/boards/native/drivers'
native-uart0.c: In function 'init_tcp_socket':
native-uart0.c:138:5: warning: 's' may be used uninitialized in this function [-Wmaybe-uninitialized]
     return s;
     ^
```

--- a/boards/native/drivers/native-uart0.c
+++ b/boards/native/drivers/native-uart0.c
@@ -92,7 +92,7 @@ void *get_in_addr(struct sockaddr *sa)
 int init_tcp_socket(char *tcpport)
 {
     struct addrinfo hints, *info, *p;
-    int i, s;
+    int i, s = -1;
     if (tcpport == NULL) {
         tcpport = UART_TCPPORT;
     }

